great_expectations/datasource/fluent/interfaces.py

```diff
@@ -170,8 +170,32 @@ def batch_parameters_to_batch_spec_kwarg_identifiers(
         ...
 
 
-class TestConnectionError(Exception):
-    pass
+class TestConnectionError(ConnectionError):
+    """
+    Raised if `.test_connection()` fails to connect to the datasource.
+    """
+
+    def __init__(
+        self,
+        message: str = "Attempt to connect to datasource failed",
+        *,
+        cause: Exception | None = None,
+        addendum: str | None = None,
+    ):
+        """
+        Args:
+            `message` base of the error message to be provided to the user.
+            `cause` is the original exception that caused the error, the repr of which will be added
+                to the error message.
+            `addendum` is optional additional information that can be added to the error message.
+        """
+        self.cause = cause  # not guaranteed to be the same as `self.__cause__`
+        self.addendum = addendum
+        if cause:
+            message += f": due to {cause!r}"
```

Comment:
```suggestion
            message += f": due to {cause.__class__.__name__}:{e!s}"
```
Alternatively, we could do this, which might look slightly better in our app.